src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -516,6 +516,89 @@ default Path resolve(String other) {
         return resolve(getFileSystem().getPath(other));
     }
 
+    /**
+     * Resolves a path or a sequence of paths against this path.
+     *
+     * <p> The result of this method is the same as would be obtained
+     * by resolving the first parameter path against this path, then
+     * resovling the second parameter path against the derived path,
+     * and so on until all parameters have been processed. Each single
+     * resolution in the chain behaves as specified by {@linkplain
+     * #resolve(Path) resolve}.
+     *
+     * @implSpec
+     * The default implementation is equivalent for this path to:
+     * {@snippet lang=java :
+     * Path result = resolve(first);
+     * for (Path p : more) {
+     *     result = result.resolve(p);
+     * }
+     * }
+     *
+     * @param first the first path to resolve against this path
+     *
+     * @param more additional paths to be joined to form the path
+     *
+     * @return the resulting path
+     *
+     * @throws  InvalidPathException
+     *          if the path string cannot be converted to a Path.
+     *
+     * @see #resolve(Path)
+     *
+     * @since 22
+     */
+    default Path resolve(Path first, Path... more) {
+        Path result = resolve(first);
+        for (Path p : more) {
```

Comment:
Hello Brian, do you think there should be a `@throws NullPointerException` in the javadoc, if any of these `Path` values is `null`?

Reply:
You're right indeed. The `package-info.java` does say:

```
   Unless otherwise noted, passing a {@code null} argument to a
 * constructor or method in any class or interface in this package
 * will cause a {@link java.lang.NullPointerException
 * NullPointerException} to be thrown.
```
Before commenting, I had checked if the class level documentation had this detail and didn't see it, but I had forgotten to check package level documentation.